src/AI/JakeMan.java

```diff
@@ -846,29 +1107,122 @@ else if( um == copter ) // Consider downgrading a tank to an inf to get a copter
+      var copterCounters = unitTypeToCounterMap.get(copter);
+      CounterRatio copterCounterCopter = new CounterRatio(copter, 200/3);
+      copterCounterCopter.roundTargetPercentUp = true; // I don't consider a single existing copter a full counter to an enemy copter
+      copterCounters.add(copterCounterCopter);
+      copterCounters.add(new CounterRatio(antiAir, 100));
+    }
+
+    // Md: 1 neo per 1.5 Mds? Ignore if you already have a bomber
+    counterOrder.add(mdTank);
+    unitTypeToCounterMap.put(mdTank, new ArrayList<>());
+    var mdCounters = unitTypeToCounterMap.get(mdTank);
+    var bomber = myArmy.cos[0].getUnitModel(UnitModel.AIR_TO_SURFACE | UnitModel.JET, false);
+    if( null != bomber )
+      mdCounters.add(new CounterRatio(bomber, 250));
+    var neoTank = allTanks.get(2);
+    if( null != neoTank )
+      mdCounters.add(new CounterRatio(neoTank, 150));
+
+    // 1 fighter per 1.7 (yes) bombers
+    //   (trunctate after the multiplicarion, so you get one in response to one bomber and a second in response to the third),
+    var fighter = myArmy.cos[0].getUnitModel(UnitModel.AIR_TO_AIR | UnitModel.JET, false);
+    if( null != bomber )
+    {
+      counterOrder.add(bomber);
+      unitTypeToCounterMap.put(bomber, new ArrayList<>());
+      var bomberCounters = unitTypeToCounterMap.get(bomber);
+      if( null != fighter )
+        bomberCounters.add(new CounterRatio(fighter, 170));
+      bomberCounters.add(new CounterRatio(antiAir, 50));
+
+      // or 2 non-copter-calc-involved AAs per bomber
+      if( null != copter )
+      {
+        counterTypeNegationMap.put(bomber, new HashMap<>());
+        var bomberCounterNegators = counterTypeNegationMap.get(bomber);
+        bomberCounterNegators.put(antiAir, new ArrayList<>());
+        var aaCounterCounters = bomberCounterNegators.get(antiAir);
+        aaCounterCounters.add(new CounterRatio(copter, 100));
+      }
+    }
+
+    // stealth: have 1 healthy fighter on the board if a stealth is present
+    var stealths = myArmy.cos[0].getAllModels(UnitModel.AIR_TO_SURFACE | UnitModel.AIR_TO_AIR | UnitModel.JET, false);
+    boolean stealthIsStealth = false;
+    for( var s : stealths )
+      if( s.hidden )
+      {
+        for( var actionType : s.baseActions )
+          if( actionType instanceof TransformLifecycle.TransformFactory )
+          {
+            stealthIsStealth = true;
+            break;
+          }
+        if( stealthIsStealth )
+          break;
+      }
+    if( stealthIsStealth && null != fighter )
+      for( var s : stealths )
+      {
+        counterOrder.add(s);
+        unitTypeToCounterMap.put(s, new ArrayList<>());
+        var stealthCounters = unitTypeToCounterMap.get(s);
+        CounterRatio fighterCounterStealth = new CounterRatio(fighter, 100);
+        fighterCounterStealth.roundTargetPercentUp = true; // Fighter needs to be healthy to deal
+        stealthCounters.add(fighterCounterStealth);
```

Comment:
I'm not sure I understand what's happening here either. It looks like you are mapping the fighter unit model as a counter to the stealth unit model, but both models are drawn from `myArmy.cos[0]` above - are we countering our own unit types?

Reply:
We are mapping from and to *universal* `UnitModel`s.
Doesn't matter if we're Kanbei or Colin, or who we're fighting - our behavior is the same, so we don't care about the specifics.